modules/ding_permissions/ding_permissions.module

```diff
@@ -1000,12 +1001,10 @@ function ding_permissions_secure_permissions($role) {
  * Implements hook_preprocess_HOOK().
  */
 function ding_permissions_preprocess_html(&$variables) {
-  global $user;
```

Comment:
@holt83: I see your point regarding careful use of permissions.

Could we avoid the preprocess altogether and instead implement this logic in `ding_permissions_secure_permissions()` programmatically as is done with `promote to front page option`?

I am thinking something like:

```php
// Ensure that administrative users only have access to one navigation system: Admin menu and not toolbar.
foreach ($permissions as $role => &$role_permissions) {
  if (in_array('access administration pages', $role_permissions)) {
    $role_permissions = array_unique(array_merge($role_permissions, array('access administration menu')));
    $role_permissions = array_diff($role_permissions, ['access toolbar']);
  }
}
```